File: clients/Ruby/model/player.rb
class Player
    attr_accessor :id
    attr_accessor :score
    attr_accessor :resource
    def initialize(id, score, resource)
        @id = id
        @score = score
        @resource = resource
    end
    def self.read_from(stream)
        id = stream.read_int()
        score = stream.read_int()
        resource = stream.read_int()
        Player.new(id, score, resource)
    end
    def write_to(stream)
        stream.write_int(@id)
        stream.write_int(@score)
        stream.write_int(@resource)
    end
end

File: clients/TypeScript/src/model/Player.ts
import {Stream} from './Stream';

export class Player {
    constructor(
        public id: number,
        public score: number, 
        public resource: number,
    ) {}

    static async readFrom(stream: Stream): Promise<Player> {
        const id = await stream.readInt();
        const score  = await stream.readInt();
        const resource = await stream.readInt();

        return new Player(id, score, resource);
    }

    async writeTo(stream: Stream) {
        await stream.writeInt(this.id);
        await stream.writeInt(this.score);
        await stream.writeInt(this.resource);
    }
}
